mlflow/pyfunc/backend.py

```diff
@@ -137,33 +230,125 @@ def copy_model_into_container(dockerfile_context_dir):
         )
 
 
-def _execute_in_conda_env(conda_env_path, command, install_mlflow, command_env=None):
+def _execute_in_conda_env(conda_env_path, command, install_mlflow, env_id, command_env=None):
     if command_env is None:
         command_env = os.environ
-    env_id = os.environ.get("MLFLOW_HOME", VERSION) if install_mlflow else None
     conda_env_name = get_or_create_conda_env(conda_env_path, env_id=env_id)
     activate_conda_env = get_conda_command(conda_env_name)
     if install_mlflow:
-        if "MLFLOW_HOME" in os.environ:  # dev version
-            install_mlflow = "pip install -e {} 1>&2".format(os.environ["MLFLOW_HOME"])
-        else:
-            install_mlflow = "pip install mlflow=={} 1>&2".format(VERSION)
+        activate_conda_env += [_get_install_mlflow_command()]
+    _logger.info("=== Running command '%s'", command)
+    _run_command(
+        [
+            *_get_entrypoint(),
+            _join_commands(
+                *activate_conda_env,
+                command,
+            ),
+        ],
+        env=command_env,
+    )
 
-        activate_conda_env += [install_mlflow]
-    if os.name != "nt":
-        separator = " && "
-    else:
-        separator = " & "
 
-    command = separator.join(activate_conda_env + [command])
-    _logger.info("=== Running command '%s'", command)
+def should_use_virtualenv():
+    return "MLFLOW_USE_VIRTUALENV" in os.environ
```

Comment:
Why not add a `mlflow models CLI` argument like "env_type" (available values `local`, `conda`, `virtualenv`) and deprecate the `no-conda` argument ?

Reply:
I'll add `--env-manager` to `mlflow models serve` command (there is a section for this in my design doc).

Reply:
oh yes.